Sources/GRPC/CallHandlers/_BaseCallHandler.swift

```diff
@@ -23,12 +23,11 @@ import Logging
 ///
 /// Calls through to `processMessage` for individual messages it receives, which needs to be implemented by subclasses.
 /// - Important: This is **NOT** part of the public API.
-public class _BaseCallHandler<RequestPayload: GRPCPayload, ResponsePayload: GRPCPayload>: GRPCCallHandler {
+public class _BaseCallHandler<Request, Response>: GRPCCallHandler {
+  private let codec: ChannelHandler
+
   public func makeGRPCServerCodec() -> ChannelHandler {
-    return HTTP1ToGRPCServerCodec<RequestPayload, ResponsePayload>(
-      encoding: self.callHandlerContext.encoding,
-      logger: self.logger
-    )
+    return self.codec
```

Comment:
This has very subtly changed behaviour: it now returns the _same_ codec everywhere, while before it returned different ones. I believe it's ok, but we need to make sure we document clearly on the codecs that they are _required_ to be stateless.

Reply:
In fact this is so unlikely to be achieved (as those codecs are generic over other things) that it would maybe be better to require that this have a codec _factory_, not a codec.

Reply:
True; this is only ever called once per RPC. We should just change the API to reflect that.